homeassistant/components/solarlog/sensor.py

```diff
@@ -206,39 +225,52 @@ async def async_setup_entry(
 ) -> None:
     """Add solarlog entry."""
     coordinator = entry.runtime_data
-    async_add_entities(
-        SolarlogSensor(coordinator, description) for description in SENSOR_TYPES
-    )
 
+    # https://github.com/python/mypy/issues/14294
+
+    entities: list[SensorEntity] = [
+        SolarLogCoordinatorSensor(coordinator, sensor)
+        for sensor in SOLARLOG_SENSOR_TYPES
+    ]
+
+    device_data = coordinator.data.get("devices", {})
+
+    if device_data != {}:
```

Comment:
```suggestion
    if not device_data:
```

Reply:
You're right, this test now also became obsolete.